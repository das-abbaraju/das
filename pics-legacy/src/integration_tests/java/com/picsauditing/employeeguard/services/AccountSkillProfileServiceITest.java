package com.picsauditing.employeeguard.services;

import com.picsauditing.employeeguard.daos.*;
import com.picsauditing.employeeguard.entities.AccountSkill;
import com.picsauditing.employeeguard.entities.AccountSkillProfile;
import com.picsauditing.employeeguard.entities.Profile;
import com.picsauditing.employeeguard.forms.employee.SkillDocumentForm;
import com.picsauditing.employeeguard.forms.factory.FormBuilderFactory;
import com.picsauditing.employeeguard.services.entity.ProfileEntityService;
import com.picsauditing.employeeguard.services.entity.SkillEntityService;
import com.picsauditing.employeeguard.services.status.SkillStatus;
import com.picsauditing.employeeguard.services.status.SkillStatusCalculator;
import com.picsauditing.employeeguard.viewmodel.model.SkillInfo;
import com.picsauditing.util.SpringUtils;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
import org.springframework.test.context.web.WebAppConfiguration;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import java.util.HashSet;
import java.util.Set;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration( {"classpath:EGITest-localhost-context.xml"} )
@WebAppConfiguration
@ActiveProfiles("jpa")
//@Transactional
public class AccountSkillProfileServiceITest {

	private static final int LENNY_LEONARD_PROFILE_ID =1;
	private static final int LENNY_LEONARD_EMPLOYEE_ID =31;
	private static final int ANOTHER_EMPLOYEE_ID =41;
	private static final int LENNY_LEONARD_EMPLOYEE_CONTRACTOR_ACCOUNT_ID =54578;
	private static final int LENNY_LEONARD_PROFILE_DOC_ID =9;
	private static final int SKILL_FirePreventionTraining_ID =12;
	private static final int SKILL_GeneralSafetyTraining_ID =13;
	private static final int PROJECT_ID =1;
	private static final int ROLE_ID =1;
	private static final int CORPORATE_ID =55653;
	private static final int SITE_ID =55654;
	private static final Set CONTRACTORS =new HashSet(){{add(LENNY_LEONARD_EMPLOYEE_CONTRACTOR_ACCOUNT_ID);}};
	private static final Set CORPORATES =new HashSet(){{add(CORPORATE_ID);}};

	@Autowired
	private AccountSkillProfileDAO accountSkillProfileDAO;
	@Autowired private ProfileDAO profileDAO;
	@Autowired private ProfileDocumentDAO profileDocumentDAO;
	@Autowired private EmployeeDAO employeeDAO;
	@Autowired private AccountSkillDAO accountSkillDAO;
	@Autowired private ProjectDAO projectDAO;
	@Autowired private RoleDAO roleDAO;

	@Autowired
	private AccountSkillProfileService accountSkillProfileService;
	@Autowired
	private ProfileEntityService profileEntityService;
	@Autowired
	private ProfileDocumentService profileDocumentService;
	@Autowired
	private SkillEntityService skillEntityService;

	@Autowired
	private FormBuilderFactory formBuilderFactory;

	@PersistenceContext
	private EntityManager em;

	public void deleteAccountSkillProfile(int id){
		AccountSkillProfile accountSkillProfile=em.find(AccountSkillProfile.class, id);
		em.merge(accountSkillProfile);
		em.remove(accountSkillProfile);
		em.flush();

	}

	@Test
	public void testUpdate_BrandNewSkillUpdate_ExpectCompletedStatus() throws Exception {
		Profile profile = profileDAO.find(LENNY_LEONARD_PROFILE_ID);
		AccountSkill accountSkill = accountSkillDAO.find(SKILL_FirePreventionTraining_ID);
		AccountSkillProfile accountSkillProfile = accountSkillProfileDAO.findByProfileAndSkill(profile,accountSkill);

		if(accountSkillProfile!=null) {
			accountSkillProfile=accountSkillProfileDAO.find(accountSkillProfile.getId());
			accountSkillProfileDAO.delete(accountSkillProfile);
		}

		SkillInfo skillInfo = formBuilderFactory.getSkillInfoBuilder().build(accountSkill, SkillStatus.Completed);
		SkillDocumentForm skillDocumentForm = formBuilderFactory.getSkillDocumentFormBuilder().build(skillInfo, null);
		skillDocumentForm.setVerified(true);

		accountSkillProfileService.update(accountSkill, profile, skillDocumentForm);

		accountSkillProfile = accountSkillProfileDAO.findByProfileAndSkill(profile,accountSkill);

		assertNotNull(accountSkillProfile);

		SkillStatus skillStatus = SkillStatusCalculator.calculateStatusFromSkill(accountSkillProfile);
		assertEquals(skillStatus.toString(),SkillStatus.Completed.toString());
	}

}
