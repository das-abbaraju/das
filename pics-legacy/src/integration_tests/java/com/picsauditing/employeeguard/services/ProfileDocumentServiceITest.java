package com.picsauditing.employeeguard.services;

import com.picsauditing.PICS.I18nCache;
import com.picsauditing.access.Permissions;
import com.picsauditing.jpa.entities.Account;
import com.picsauditing.web.SessionInfoProvider;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

import static org.mockito.Mockito.when;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration( {"classpath:EGITest-localhost-context.xml"} )
public class ProfileDocumentServiceITest {

	@Mock
	private Permissions permissions;

	@Mock
	private SessionInfoProvider sessionInfoProvider;

	@Autowired
	private ProfileDocumentService profileDocumentService;

	private static final int SKILL_ID = 11;
	private static final int DOCUMENT_ID = 9;
	private static final int APP_USER_ID = 131073;

	@Before
	public void setUp() throws Exception {
		MockitoAnnotations.initMocks(this);
	}

	@Test
	public void testUpdate_getDocumentThumbnail_CheckEmployeePermissions() throws Exception {
		when(sessionInfoProvider.getAppUserId()).thenReturn(APP_USER_ID);
		profileDocumentService.getAccountSkillProfileForEmployeeAndSkill(SKILL_ID, DOCUMENT_ID);
	}

}
