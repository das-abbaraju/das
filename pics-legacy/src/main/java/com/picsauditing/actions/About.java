package com.picsauditing.actions;

import java.text.SimpleDateFormat;
import java.util.Date;

import com.picsauditing.access.Anonymous;
import com.picsauditing.access.UserAgentParser;

public class About extends PicsActionSupport {

	private String browserName;
	private String operatingSystem;
	private Date systemTime;

	private static final long serialVersionUID = 1L;

	@Override
	@Anonymous
	public String execute() {
		UserAgentParser userAgentParser = new UserAgentParser(getRequest().getHeader("User-Agent"));
		browserName = userAgentParser.getBrowserName();
		operatingSystem = userAgentParser.getBrowserOperatingSystem();

		return SUCCESS;
	}

	public String getBrowserName() {
		return browserName;
	}

	public String getOperatingSystem() {
		return operatingSystem;
	}

	public String getSystemTime() {
		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
		systemTime = new Date();

		return dateFormat.format(systemTime);
	}
}