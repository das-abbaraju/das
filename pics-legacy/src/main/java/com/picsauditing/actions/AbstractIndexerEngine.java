package com.picsauditing.actions;

import com.picsauditing.dao.IndexableDAO;
import com.picsauditing.jpa.entities.*;
import com.picsauditing.search.Database;
import com.picsauditing.util.IndexerEngine;
import com.picsauditing.util.ReflectUtil;
import com.picsauditing.util.Strings;
import org.springframework.beans.factory.annotation.Autowired;

import java.sql.SQLException;
import java.util.HashSet;
import java.util.Set;

public abstract class AbstractIndexerEngine implements IndexerEngine {

    @Autowired
	protected IndexableDAO indexableDao;

	protected Database db = new Database();

	private static Set<Class<? extends Indexable>> entries = new HashSet<Class<? extends Indexable>>();

	static {
		entries.add(Account.class);
		entries.add(User.class);
		entries.add(Employee.class);
		entries.add(Trade.class);
	}

	@Override
	public void updateIndex(int id, Class<? extends Indexable> clazz) {
		Set<Integer> saved = new HashSet<Integer>();
		saved.add(id);
		updateIndex(saved, clazz);
	}

	@Override
	public void updateIndex(Set<Integer> saved, Class<? extends Indexable> clazz) {
		if (saved.isEmpty())
			return;
		String updateIndexing = "UPDATE " + ReflectUtil.getTableName(clazz) + " SET needsIndexing=0 WHERE id IN ("
				+ Strings.implode(saved) + ")";
		try {
			db.executeUpdate(updateIndexing);
		} catch (SQLException e) {
			// TODO Auto-generated catch block
		}
	}

	public void setIndexableDao(IndexableDAO indexableDao) {
		this.indexableDao = indexableDao;
	}

	public Set<Class<? extends Indexable>> getEntries() {
		return entries;
	}

}
