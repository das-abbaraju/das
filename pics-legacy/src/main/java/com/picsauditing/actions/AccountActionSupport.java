package com.picsauditing.actions;

import com.picsauditing.access.OpPerms;
import com.picsauditing.dao.AccountDAO;
import com.picsauditing.dao.CountryDAO;
import com.picsauditing.dao.CountrySubdivisionDAO;
import com.picsauditing.dao.NoteDAO;
import com.picsauditing.jpa.entities.*;
import com.picsauditing.model.contractor.AccountContact;
import com.picsauditing.util.Strings;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.*;

@SuppressWarnings("serial")
public class AccountActionSupport extends PicsActionSupport {

	protected int id;
	// protected Account account;
	protected String subHeading = null;
	// TODO This local-copy optimzation is probably unnecessary, being redundant
	// with Hibernate's caching.
	protected List<Note> notes;
	protected NoteCategory noteCategory = NoteCategory.General;
	protected int questionId;

	@Autowired
	protected AccountDAO accountDAO;
	@Autowired
	private CountrySubdivisionDAO countrySubdivisionDAO;

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public int getQuestionId() {
		return questionId;
	}

	public void setQuestionId(int questionId) {
		this.questionId = questionId;
	}

	/**
	 * Delete this method and property when we start i18n
	 * 
	 * @return
	 */
	public String getSubHeading() {
		return subHeading;
	}

	public void setSubHeading(String subHeading) {
		this.subHeading = subHeading;
	}

	public void setAccount(Account account) {
		this.account = account;
	}

	public Account getAccount() {
		if (account == null) {
			loadPermissions();
			account = accountDAO.find(permissions.getAccountId(), permissions.getAccountType());
		}
		return account;
	}

	/************* NOTES ************/
	@Deprecated
	protected NoteDAO getNoteDao() {
		return noteDao;
	}

	public NoteCategory getNoteCategory() {
		return noteCategory;
	}

	public void setNoteCategory(NoteCategory noteCategory) {
		this.noteCategory = noteCategory;
	}

	/**
	 * Get a list of notes up to the limit, using the given where clause
	 * 
	 * @param where
	 *            should be in the format of "AND field=1", can be an empty
	 *            string
	 * @param limit
	 *            e.g. 25
	 * @return
	 */
	public List<Note> getNotes(String where, int firstLimit, int limit) {
		if (notes == null) {
			notes = noteDao.getNotes(id, permissions, "status IN (1,2)" + where, firstLimit, limit);
		}

		return notes;
	}

	/**
	 * Using the same parameters as getNotes(), above, see if there is at least
	 * one more note that can be fetched after that.
	 */
	public boolean atLeastOneMoreNote(String where, int firstLimit, int limit) {
		List<Note> zeroOrOneMoreNote = noteDao.getNotes(id, permissions, "status IN (1,2)" + where, firstLimit + limit,
				1);
		return (zeroOrOneMoreNote.size() > 0);
	}

	/**
	 * Get a list of 5 embedded notes, based on noteCategory
	 * 
	 * @return
	 */
	public List<Note> getNotes() {
		return getNotes(" AND noteCategory IN ('" + noteCategory.toString() + "','General')", 0, 5);
	}

	protected Note addNote(Account account, String newNote) {
		return addNote(account, newNote, noteCategory);
	}

	protected Note addNote(Account account, String newNote, NoteCategory noteCategory) {
		return addNote(account, newNote, noteCategory, LowMedHigh.Low, true, Account.EVERYONE, null, null);
	}

	protected Note addNote(Account account, String newNote, NoteCategory noteCategory, int viewableBy) {
		return addNote(account, newNote, noteCategory, LowMedHigh.Low, true, viewableBy, null, null);
	}

	protected Note addNote(Account account, String newNote, NoteCategory noteCategory, LowMedHigh priority,
			boolean canContractorView, int viewableBy, User user) {
		return addNote(account, newNote, noteCategory, LowMedHigh.Low, true, viewableBy, user, null);
	}

    protected Note addNote(Account account, String summary, String newNote, NoteCategory category, LowMedHigh priority,
                           boolean canContractorView, int viewableBy) {
        Note note = new Note();
        note.setAccount(account);
        note.setAuditColumns(permissions);
        note.setSummary(summary);
        note.setBody(newNote);
        note.setPriority(priority);
        note.setNoteCategory(category);
        note.setViewableById(viewableBy);
        note.setCanContractorView(canContractorView);
        note.setStatus(NoteStatus.Closed);
        noteDao.save(note);
        return note;
    }

    protected Note addNote(Account account, String newNote, NoteCategory category, LowMedHigh priority,
			boolean canContractorView, int viewableBy, User user, Employee employee) {
		Note note = new Note();
		note.setAccount(account);
		note.setAuditColumns(permissions);
		note.setSummary(newNote);
		note.setPriority(priority);
		note.setNoteCategory(category);
		note.setViewableById(viewableBy);
		note.setCanContractorView(canContractorView);
		note.setStatus(NoteStatus.Closed);
		note.setEmployee(employee);
		noteDao.save(note);
		return note;
	}

	public int getViewableByAccount(Account account) {
		if (account != null) {
			return account.getId();
		}
		return Account.EVERYONE;
	}

	/***** END of NOTES *****/

	@Deprecated
	public CountryDAO getCountryDAO() {
		return countryDAO;
	}

	@Deprecated
	public CountrySubdivisionDAO getCountrySubdivisionDAO() {
		return countrySubdivisionDAO;
	}

	public CountrySubdivision getAccountCountrySubdivision() {
		CountrySubdivision value = null;
		Account account = accountDAO.find(id, "Contractor");
		if (account != null) {
			value = account.getCountrySubdivision();
		}
		return value;
	}

	public String getCountrySubdivisionString() {
		String value = "";
		Account account = accountDAO.find(id, "Contractor");
		if (account != null) {
			value = account.getCountrySubdivision().getSimpleName();
		}
		return value;
	}

	public List<Country> getCountryList() {
		List<Country> countryList = countryDAO.findAll();
		Collections.sort(countryList, Country.NAME_COMPARATOR);
		countryList.add(0, countryDAO.find("GB"));
		countryList.add(0, countryDAO.find("CA"));
		countryList.add(0, countryDAO.find("US"));

		return countryList;
	}

	public List<CountrySubdivision> getCountrySubdivisionList() {
		if (account == null) {
			return getCountrySubdivisionList(null);
		}

		return getCountrySubdivisionList(account.getCountry().getIsoCode());
	}

	public List<CountrySubdivision> getCountrySubdivisionList(String countries) {
		List<CountrySubdivision> result;
		if (countries == null) {
			result = countrySubdivisionDAO.findByCountries(Country.COUNTRIES_WITH_SUBDIVISIONS, false);
		} else {
			boolean negative = false;
			if (countries.startsWith("!")) {
				countries = countries.replace("!", "");
				negative = true;
			}

			List<String> validCountriesWithSubdivisions = new ArrayList<>();
			for (String country : countries.split("[|]")) {
				if (Country.COUNTRIES_WITH_SUBDIVISIONS.contains(country)) {
					validCountriesWithSubdivisions.add(country);
				}
			}

			result = countrySubdivisionDAO.findByCountries(validCountriesWithSubdivisions, negative);
		}

		Collections.sort(result, CountrySubdivision.NAME_COMPARATOR);

		return result;
	}

	public String getCountrySubdivisionLabelFor(String country) {
		return getText(getCountrySubdivisionLabelKeyFor(country));
	}

	public String getCountrySubdivisionLabelKeyFor(String country) {
		if (Strings.isNotEmpty(country)) {
			String countrySubdivisionLabelKey = String.format(Country.COUNTRY_SUBDIVISION_LABEL_FORMAT, country);

			if (hasKey(countrySubdivisionLabelKey)) {
				return countrySubdivisionLabelKey;
			}
		}

		return Country.DEFAULT_COUNTRY_SUBDIVISION_LABEL;
	}

	public boolean isShowMoreNotes() {
		if (Account.CONTRACTOR_ACCOUNT_TYPE.equals(account.getType())) {
			return permissions.hasPermission(OpPerms.ContractorDetails);
		}

		return true;
	}

	public AccountStatus[] getStatusList() {
		return AccountStatus.values();
	}

    public String getAccountLastLogin() {
        Date lastLogin = accountDAO.findLastAccountLogin(account.getId());
        return getFuzzyDate(lastLogin);
    }

    public List<AccountContact> getAccountContactsByRole(OpPerms opPerms) {
        List<AccountContact> results = accountDAO.findAccountContactsByRole(account.getId(), opPerms);
        return results;
    }
}
