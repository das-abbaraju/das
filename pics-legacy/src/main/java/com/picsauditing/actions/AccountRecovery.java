package com.picsauditing.actions;

import com.picsauditing.PICS.Grepper;
import com.picsauditing.access.Anonymous;
import com.picsauditing.authentication.entities.AppUser;
import com.picsauditing.authentication.service.AppUserService;
import com.picsauditing.dao.UserDAO;
import com.picsauditing.employeeguard.entities.Profile;
import com.picsauditing.employeeguard.services.entity.ProfileEntityService;
import com.picsauditing.jpa.entities.User;
import com.picsauditing.mail.EmailBuildErrorException;
import com.picsauditing.service.email.AccountRecoveryEmailService;
import com.picsauditing.util.Strings;
import com.picsauditing.util.URLUtils;
import com.picsauditing.validator.InputValidator;
import org.springframework.beans.factory.annotation.Autowired;

import java.io.IOException;
import java.util.Date;
import java.util.List;

@SuppressWarnings("serial")
public class AccountRecovery extends PicsActionSupport {
	@Autowired
	private UserDAO userDAO;
    @Autowired
    private AppUserService appUserService;
    @Autowired
    private ProfileEntityService profileService;
	@Autowired
	private InputValidator inputValidator;
    @Autowired
    private AccountRecoveryEmailService emails;

	private String email, username;
	private URLUtils urlUtils;

    private AccountRecoveryMessages messages = new AccountRecoveryMessages(this);

    private static final String PASSWORD = "password";
    private static final String USERNAME = "username";

    private static final Grepper<User> recoverableUserFilter = new Grepper<User>() {
        @Override
        public boolean check(User user) {
            return user != null && !user.isDeleted();
        }
    };

	@Anonymous
	@Override
	public String execute() throws Exception {
		return PASSWORD;
	}

	@Anonymous
	public String recoverUsername() {
		return USERNAME;
	}

	@Anonymous
	public String findName() throws Exception {

        if (!inputValidator.validateEmail(email).equals(InputValidator.NO_ERROR)) {
            messages.addNoUserNameError();
			return USERNAME;
		}

		final List<User> matchingUsers = recoverableUserFilter.grep(userDAO.findByEmail(email.trim()));

		if (matchingUsers.isEmpty()) {
            messages.addUserEmailNotFoundError();
            return accountRecoveryRedirect();
		} else {
            try {
                emails.sendUsernameRecoveryEmail(matchingUsers);
                messages.addUsernameEmailSentMessage();
                return loginRedirect();
            } catch (Exception e) {
                messages.addNoEmailSentError();
                return USERNAME;
            }
        }
	}



    private String loginRedirect() throws IOException {
        String loginPage = urlUtils().getActionUrl("Login");
        return setUrlForRedirect(loginPage);
    }

    private String accountRecoveryRedirect() throws IOException {
        String accountRecoveryPage = urlUtils().getActionUrl("AccountRecovery", "recoverUsername");
        return setUrlForRedirect(accountRecoveryPage);
    }

	@Anonymous
	public String resetPassword() {
		if (Strings.isEmpty(username) || username.startsWith("DELETE-")) {
            messages.addBlankUsernameSubmittedError();
			return PASSWORD;
		}

        final AppUser appUser = appUserService.findByUsername(username);

        if (appUser == null) {
            messages.addUsernameNotFoundError();
        } else {
            final User user = userDAO.findName(username);
            final Profile profile = profileService.findByAppUserId(appUser.getId());

            if (user == null && profile == null) {
                messages.addNoValidAccountFound();
            } else if (!user.isActiveB() && profile == null) {
                messages.addUserNotActiveError();
            } else {
                try {
                    addResetHashTo(appUser);
                    appUserService.save(appUser);

                    if (user != null) {
                        emails.sendRecoveryEmail(user, getRequestHost());
                        messages.successfulEmailSendTo(user.getEmail());
                    } else {
                        emails.sendRecoveryEmail(profile, getRequestHost());
                        messages.successfulEmailSendTo(profile.getEmail());
                    }

                    return setUrlForRedirect("Login.action");

                } catch (EmailBuildErrorException e) {
                    messages.failedEmailSend();
                } catch (IOException e) {
                    messages.addUsernameNotFoundError();
                }
            }
        }
        return PASSWORD;
	}

    private void addResetHashTo(AppUser appUser) {
        appUser.setResetHash(Strings.hashUrlSafe("u" + appUser.getId() + String.valueOf(new Date().getTime())));
    }

    public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getUsername() {
		return username;
	}

	public void setUsername(String username) {
		this.username = username;
	}


	private URLUtils urlUtils() {
		if (urlUtils == null) {
			urlUtils = new URLUtils();
		}
		return urlUtils;
	}


}