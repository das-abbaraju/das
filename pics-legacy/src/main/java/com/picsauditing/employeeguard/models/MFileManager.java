package com.picsauditing.employeeguard.models;

import com.google.gson.annotations.Expose;
import com.picsauditing.PICS.DateBean;
import com.picsauditing.employeeguard.entities.ProfileDocument;
import com.picsauditing.util.PicsDateFormat;

import java.util.*;

public class MFileManager {

	private Map<Integer, MFileManager.MFile> lookup = new HashMap<>();

	public static Set<MFile> newCollection() {
		return new HashSet<>();
	}

	public MFile attachWithModel(ProfileDocument profileDocument) {
		int id = profileDocument.getId();

		if (lookup.get(id) == null) {
			lookup.put(id, new MFile(profileDocument));
		}

		return lookup.get(id);
	}

	public Set<MFile> copyBasicInfo(List<ProfileDocument> profileDocuments) {
		Set<MFile> mFiles = MFileManager.newCollection();
		for (ProfileDocument profileDocument : profileDocuments) {
			MFile mFile = this.attachWithModel(profileDocument);
			mFile.copyId().copyName().copyCreatedDate().copyExpirationDate();
			mFiles.add(mFile);
		}

		return mFiles;
	}

	public static class MFile {

		@Expose
		private Integer id;
		@Expose
		private String name;
		@Expose
		private String createdDate;
		@Expose
		private String expirationDate;

		private ProfileDocument profileDocument;

		public MFile(ProfileDocument profileDocument) {
			this.profileDocument = profileDocument;
		}

		public MFile copyId() {
			id = profileDocument.getId();
			return this;
		}

		public MFile copyName() {
			name = profileDocument.getName();
			return this;
		}

		public MFile copyCreatedDate() {
			this.createdDate = PicsDateFormat.formatDateIsoOrBlank(profileDocument.getCreatedDate());
			return this;
		}

		public MFile copyExpirationDate() {
			this.expirationDate = DateBean.getEndOfTime().equals(profileDocument.getEndDate()) ? "Never" :
					PicsDateFormat.formatDateIsoOrBlank(profileDocument.getEndDate());
			return this;
		}

		public int getId() {
			return id;
		}

		public void setId(int id) {
			this.id = id;
		}

		public String getName() {
			return name;
		}

		public void setName(String name) {
			this.name = name;
		}

		public String getCreatedDate() {
			return createdDate;
		}

		public void setCreatedDate(String createdDate) {
			this.createdDate = createdDate;
		}

		public String getExpirationDate() {
			return expirationDate;
		}

		public void setExpirationDate(String expirationDate) {
			this.expirationDate = expirationDate;
		}

		@Override
		public boolean equals(Object o) {
			if (this == o) return true;
			if (o == null || getClass() != o.getClass()) return false;

			MFile mFile = (MFile) o;

			if (profileDocument != null ? !profileDocument.equals(mFile.profileDocument) : mFile.profileDocument != null)
				return false;

			return true;
		}

		@Override
		public int hashCode() {
			return profileDocument != null ? profileDocument.hashCode() : 0;
		}
	}
}
