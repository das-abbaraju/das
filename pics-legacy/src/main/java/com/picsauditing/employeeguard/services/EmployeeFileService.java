package com.picsauditing.employeeguard.services;

import com.picsauditing.employeeguard.models.MFileManager;
import com.picsauditing.employeeguard.services.entity.DocumentEntityService;
import org.springframework.beans.factory.annotation.Autowired;

import java.util.Set;

public class EmployeeFileService {

	@Autowired
	private DocumentEntityService documentEntityService;

	public Set<MFileManager.MFile> findEmployeeFiles(final int appUserId) {
		return new MFileManager().copyBasicInfo(documentEntityService.findDocumentsForAppUser(appUserId));
	}
}
