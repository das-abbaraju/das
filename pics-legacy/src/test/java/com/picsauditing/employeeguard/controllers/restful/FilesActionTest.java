package com.picsauditing.employeeguard.controllers.restful;

import com.picsauditing.PicsActionTest;
import com.picsauditing.actions.PicsActionSupport;
import com.picsauditing.employeeguard.models.MFileManager;
import com.picsauditing.employeeguard.services.EmployeeFileService;
import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.powermock.reflect.Whitebox;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

import static com.picsauditing.employeeguard.EGTestDataUtil.PROFILE_DOCUMENT_1;
import static com.picsauditing.employeeguard.EGTestDataUtil.PROFILE_DOCUMENT_2;
import static org.junit.Assert.assertEquals;
import static org.mockito.Matchers.anyInt;
import static org.mockito.Mockito.when;

@UseReporter(DiffReporter.class)
public class FilesActionTest extends PicsActionTest {

	@Mock
	private EmployeeFileService employeeFileService;

	// Class under test
	private FilesAction filesAction;

	@Before
	public void setUp() throws Exception {
		MockitoAnnotations.initMocks(this);

		filesAction = new FilesAction();

		super.setUp(filesAction);

		Whitebox.setInternalState(filesAction, "employeeFileService", employeeFileService);
	}

	@Test
	public void testIndex() throws Exception {
		setupTestIndex();

		String result = filesAction.index();

		verifyTestIndex(result);
	}

	private void setupTestIndex() {
		Set<MFileManager.MFile> fakeMFiles = new HashSet<>(Arrays.asList(
				new MFileManager.MFile(PROFILE_DOCUMENT_1).copyId().copyName().copyCreatedDate().copyExpirationDate(),
				new MFileManager.MFile(PROFILE_DOCUMENT_2).copyId().copyName().copyCreatedDate().copyExpirationDate()));
		when(employeeFileService.findEmployeeFiles(anyInt())).thenReturn(fakeMFiles);
	}

	private void verifyTestIndex(String result) throws Exception {
		assertEquals(PicsActionSupport.JSON_STRING, result);
		Approvals.verify(filesAction.getJsonString());
	}
}
