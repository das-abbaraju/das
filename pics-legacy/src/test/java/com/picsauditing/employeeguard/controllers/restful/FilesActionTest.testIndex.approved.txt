[{"id":1,"name":"Document 1","createdDate":1391198400000,"expirationDate":64060703999000,"verificationDate":1391198400000,"neverExpires":true},{"id":2,"name":"Document 2","createdDate":1359662400000,"expirationDate":1425283200000,"neverExpires":false}]